### DISCUSSION TITLE
使用自定义 S3  endpoint 时无法指定 region 导致上传失败

### DISCUSSION THREAD

**Discussion by looching** [gee1k/uPic#248] [General]
比如 cloudflare r2

**Comment by LyonLeopard** [comment]
<img width="1019" alt="image" src="https://github.com/user-attachments/assets/f96e6ae7-028d-42c4-8bd1-e109883cea61">
There's no to set the region name when I try to use a 3rd party s3 like Cloudflare R2, which results in failure.
Uploading to R2 is a super important feature for me. I'm using the homebrew version, so idk whether you fixed this in app store version. If so, I would be happy to subscribe to your product.

**Reply by shiny** [reply]
To address compatibility issues with Cloudflare R2, I developed an API that allows adding custom uploads to Cloudflare R2. You can refer to the documentation at [https://lee.netcup.gifts/lee/docs/opensource/upic](https://lee.netcup.gifts/lee/docs/opensource/upic).

![https://img.netcup.gifts/blog/202503/FIIgku.jpg](https://img.netcup.gifts/blog/202503/FIIgku.jpg)

**Reply by shiny** [reply]
If needed, I can add other adapters as long as it provide API support. If anyone is interested, feel free to let me know. Project repository: [https://github.com/shiny/upic-custom-uploader](https://github.com/shiny/upic-custom-uploader).
